frappe/model/base_document.py

```diff
@@ -152,8 +152,9 @@ def update(self, d):
 		if "name" in d:
 			self.name = d["name"]
 
+		ignore_children = hasattr(self, "flags") and self.flags.ignore_children
```

Comment:
We use `self.flags` in a lot of places in `BaseDocument`. Expect it?

The `BaseDocument` class is being directly used only in `meta.py` - we can just use `Document` there instead?